libvast/src/plugin.cpp

```diff
@@ -353,7 +353,9 @@ caf::expected<store_actor_plugin::builder_and_header>
 store_plugin::make_store_builder(system::accountant_actor accountant,
                                  system::filesystem_actor fs,
                                  const vast::uuid& id) const {
-  auto store = make_active_store();
+  const auto& vast_config = caf::get<caf::settings>(
+    content(fs->home_system().config()).find("vast")->second);
```

Comment:
Does this still work if I provide no config at all, or do you de-reference a nullptr then? `find(...)->...` looks rather unsafe to me.

Similarly, what happens if it's a string and not a dictionary? Doesn't `caf::get<caf::settings>` just run into undefined behavior (or an assertion) then?

Since the configuration is user input we need to access it with rather defensive mechanisms.

Reply:
```
  const auto& vast_config
    = caf::get_or(content(fs->home_system().config()), "vast", caf::settings{});
```

Is probably better, which yields a default / empty settings in case there's nothing called `vast` in there. `caf::get_or` also does the right thing if the config option is not itself a `caf::settings`, so I hope that is sufficiently safe.

Reply:
You could also use `caf::get_if`, which is able to correctly distinguish between "not provided" and "not provided."

I think for this particular one, there's actually an overload of `caf::get_*` that you can use to step through the nesting directly:

```
caf::get_or(settings, "vast.zstd-compression-level", default_zstd_compression_level)
```

Reply:
Indeed, `caf::get_or` also just usees `get_if` under the hood.

The question here is what parts of the config we want to expose to the module. At the moment, the construct above confines the config that is exposed to plugins to the `vast` namespace, which I think is the correct thing to do, as I believe a plugin shouldn't directly deal with `caf` config options.

Reply:
I find this to be very problematic from a DX perspective, because you then can no longer grep for the fully qualified config options. If you pass in a large settings object like this you might as well just pass in everything.

Reply:
Modified accordingly. We now pass in the entire config, and access `vast.zstd-compression-level` in the plugin.